src/test/shared/sam/cli/samCliInvoker.test.ts

```diff
@@ -5,18 +5,83 @@
 
 'use strict'
 
+import * as assert from 'assert'
+import { Stats } from 'fs'
+import { TestLogger } from '../../../../shared/loggerUtils'
 import { SamCliConfiguration } from '../../../../shared/sam/cli/samCliConfiguration'
-import { DefaultSamCliProcessInvoker } from '../../../../shared/sam/cli/samCliInvoker'
+import {
+    DefaultSamCliProcessInvoker,
+    makeSamCliProcessInvokerContext
+} from '../../../../shared/sam/cli/samCliInvoker'
 import { assertRejects } from '../../utilities/assertUtils'
 
 describe('DefaultSamCliInvoker', async () => {
+
+    let logger: TestLogger
+
+    before( async () => {
+        logger = await TestLogger.createTestLogger()
+    })
+
+    after(async () => {
+        await logger.cleanupLogger()
+    })
+
     it('throws if sam cli location is not known', async () => {
-        const config: SamCliConfiguration = {
-            getSamCliLocation: () => undefined
-        } as any as SamCliConfiguration
+        const context = makeSamCliProcessInvokerContext({
+            cliConfig: {
+                getSamCliLocation: () => undefined
+            } as any as SamCliConfiguration
+        })
 
-        const invoker = new DefaultSamCliProcessInvoker(config)
+        const invoker = new DefaultSamCliProcessInvoker(context)
 
         await assertRejects(async () => await invoker.invoke())
     })
+
+    it('returns an error if the AWS Toolkit is out of date', async () => {
+        const testHighLevel = '999999.9999.999999'
+        const testDate = new Date(12345)
+        const testStat = new Stats()
+        testStat.mtime = testDate
+
+        const context = makeSamCliProcessInvokerContext({
+            cliConfig: {
+                getSamCliLocation: () => 'filler'
+            } as any as SamCliConfiguration,
+            cliUtils: {
+                stat: async () => testStat
+            },
+            cliInfo: { info: { version: testHighLevel }, lastModified: testDate }
+        })
+
+        const invoker = new DefaultSamCliProcessInvoker(context)
+
+        const result = await invoker.invoke()
+        assert.strictEqual(result.exitCode, 1)
+        assert.strictEqual(result.stderr, 'AWS Toolkit is out of date')
```

Comment:
What if SAM updates, but we haven't published a new version of the toolkit yet? Should we instruct users to downgrade SAM CLI?

Reply:
Very good question. I have no idea! We should probably discuss this one at some point... We could maybe prompt the user to see if they want to give the command a shot?

Reply:
Conversations with the JetBrains toolkit team indicate that we should break on advanced versions (it's better to favor safety in case the new SAM CLI has breaking changes).